### PR TITLE
Turn graphene on by default

### PR DESCRIPTION
**Pull request by ptschip**
This also required a couple of edge cases to be fixed in the request manager which were giving us trouble in the python tests.

Still needs work adding test cases to the request manager if possible. Also this introduces  a performance issue since now we have the check for if we have peers very frequently which lock cs_vNodes; I'll add an atomic flag in there so we don't have to cycle through vNodes so many times.

### CODE REVIEW

File: src/requestManager.cpp [hunk 1]
```diff
@@ -515,85 +515,90 @@ bool CRequestManager::RequestBlock(CNode *pfrom, CInv obj)
...
+
     // Ask for XTHIN's if Graphene is not enabled, or, ask for XTHIN's if graphene is enabled
     // but the grapheneblock timer has lapsed.
-    if ((IsThinBlocksEnabled() && IsChainNearlySyncd() && !IsGrapheneBlockEnabled()) ||
-        (IsThinBlocksEnabled() && IsChainNearlySyncd() && IsGrapheneBlockEnabled() &&
-            !graphenedata.CheckGrapheneBlockTimer(obj.hash)))
+    if (IsChainNearlySyncd() && IsThinBlocksEnabled() && HaveThinblockNodes())
     {
-        if (HaveThinblockNodes() && thindata.CheckThinblockTimer(obj.hash))
+        if (!IsGrapheneBlockEnabled() || (IsGrapheneBlockEnabled() && !graphenedata.CheckGrapheneBlockTimer(obj.hash)))
```
Comment (gubatron) [inline src/requestManager.cpp]:
```cpp
if (!IsGrapheneBlockEnabled() || !graphenedata.CheckGrapheneBlockTimer(obj.hash))
```
redundant logic here, a single `IsGrapheneBlockEnabled()` check is sufficient.

Comment (sickpig) [reply src/requestManager.cpp]: @gubatron 

think you are right. good catch. :smile:

Comment (ptschip) [reply src/requestManager.cpp]: ok removed the redundant IsGrapheneBlockEnabled()

**Comment by ptschip** [comment]
Taking out of WIP. Main issue was getting the request manager tests working to check all the code paths in RequestBlock()
